tsdb/querier.go

```diff
@@ -96,12 +96,12 @@ func (q *blockBaseQuerier) Close() error {
 	if q.closed {
 		return errors.New("block querier already closed")
 	}
-	var merr tsdb_errors.MultiError
-	merr.Add(q.index.Close())
-	merr.Add(q.chunks.Close())
-	merr.Add(q.tombstones.Close())
-	q.closed = true
-	return merr.Err()
+	defer func() { q.closed = true }()
```

Comment:
Why did you change this logic?

Reply:
Logic is exactly the same I believe :thinking:

Reply:
It feels a bit weird to use a defer like this, it'd be better to make it explicit I think.

Reply:
:+1: